syntax = "proto3";

package narrator.common;

message SlotDefinition {
  string name = 1;
  string type = 2;
  bool required = 3;
  string description = 4;
  bool is_entity_ref = 5;
  string entity_type = 6;
  string default_value = 7;
}

message Template {
  string id = 1;
  string name = 2;
  string title = 3;
  string description = 4;
  string category = 5;
  repeated string keywords = 6;
  repeated SlotDefinition slots = 7;
}
